### ISSUE TITLE
JSON decoding error

### ISSUE THREAD

**Issue by martinvisser** [wiremock/wiremock#2609]
### Proposal

After upgrading to WireMock 3.4.0 a previously successful test now suddenly fails with the following exception:
`org.springframework.core.codec.DecodingException: JSON decoding error: Unexpected character ('A' (code 65)): was expecting comma to separate Object entries`.

It seems to be that escaping of a double quote goes wrong now. There was no issue with version `3.3.1` or below.

Using `wiremock-standalone` version `3.4.0`.

### Reproduction steps

Using the following mapping:
```json
{
    "priority": 1,
    "request": {
        "method": "POST",
        "urlPath": "/some/path"
    },
    "response": {
        "status": 200,
        "headers": {
            "Content-Type": "application/json"
        },
        "jsonBody": {
            "data": {
                "error": "Invalid input string: \"ABC\""
            }
        }
    }
}
```

**Comment by leeturner** [comment]
Hi , thank you for reporting the issue.  I have just tried out your mapping in WireMock standalone and the response comes back as the following:

```
{
  "data": {
    "error": "Invalid input string: "
    ABC
    ""
  }
}
```

This ties into the error message you got back with regards to the Unexpected character 'A'

**Comment by tomakehurst** [comment]
Fixed in d68e984a138d3608149229b92e763a2c21362846 / 3.4.1 by rolling back the original commit that caused it.
